packages/platform-app/src/platformApps.ts

```diff
@@ -44,10 +44,18 @@ export type PlatformAppSimpleBlock =
 export type PlatformAppSettings = PlatformAppSimpleBlock | DynamicSettingBlock<AppBridgePlatformApp>;
 
 export type PlatformAppSettingsStructureExport = { [customSectionName: string]: PlatformAppSettings[] };
+export type PlatformAppMultiSettingsStructureExport = {
+    default: PlatformAppSettingsStructureExport;
+    automation?: {
+        actions?: {
+            [customActionId: string]: PlatformAppSettingsStructureExport;
+        };
+    };
+};
 
 export type PlatformAppConfigExport = {
     app: FC;
-    settings: PlatformAppSettingsStructureExport;
+    settings: PlatformAppMultiSettingsStructureExport | PlatformAppSettingsStructureExport;
```

Comment:
Looks like a lot of automation specific logic, how about just doing a new Type: "automation"? was there any discussion for that? Then we wouldnt need to adjust the settings structure nor adjust the web-app implementation and could use the default settings in it. Wouldnt the only downside be that then someone would need to have 2 Apps for 2 different functionalities?

Reply:
That makes a lot of sense for sure, thanks for going into details about it. 

My follow up question is, how would you write documentation for the automations and would you make the destinctions between what an app can do and what an automation can do?

Reply:
Regarding the downsides:
1: If i understand it correctly, what is behind an Automation action is actions that correspond to the IDs in another system and not inside of an App. To enable a new action doesn't depend on the App, but on the third party system.  When we then would like to use the output of an automation in an App, we need to define somehow code that we can run in isolation somewhere. This code will not be the same as the visual Code that is used in an App. This would then create another addition to the App structure. If we would add a new type, we could use the settings for settings, and the src folder later on for executable code that we want to run somewhere. 

2 & 3 Im not sure about this, we are aiming to have the Marketplace as a lively place, to have more Apps in there sounds like the way to go.

4. Downside of one app to rule all is that for a developer its not clear what he has to do. We mix different concepts and introduce implicit rules on what goes together and what not. For example, when you build an automation you cannot use methods, or when building an App you shouldn't use the automation part in the mainfest.json because thats for the automation usecase. It makes the communication more difficult because we mix different concepts.